### ISSUE TITLE
SubscriptionOptions is difficult to implement

### ISSUE THREAD

**Issue by PachowStudios** [nestjs/graphql#265]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

Because the functions on SubscriptionOptions have generic type parameters, the argument types cannot be asserted without causing an error.

```typescript
@Subscription(() => Message, {
  resolve: (v: Message) => v,
})
```

Causes the following error:
```
src/features/messages/messages.resolver.ts:18:32 - error TS2345: Argument of type '{ resolve: (v: Message) => Message; }' is not assignable to parameter of type 'DecoratorTypeOptions & DescriptionOptions & DepreciationOptions & SchemaNameOptions & SubscriptionOptions'.
  Type '{ resolve: (v: Message) => Message; }' is not assignable to type 'SubscriptionOptions'.
    Types of property 'resolve' are incompatible.
      Type '(v: Message) => Message' is not assignable to type '<TPayload = any, TArgs = any, TContext = any, TInfo = any, TReturnValue = any>(payload: TPayload, args: TArgs, context: TContext, info: TInfo) => TReturnValue'.
        Types of parameters 'v' and 'payload' are incompatible.
          Type 'TPayload' is not assignable to type 'Message'.

18   @Subscription(() => Message, {
                                  ~
19     resolve: (v: Message) => v,
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
20   })
```

## Expected behavior

The argument types of the resolve and filter functions should be able to be asserted in the function definition.

Defining SubscriptionOptions as follows would fix this issue:
```typescript
export interface SubscriptionOptions {
    filter?: (payload: any, variables: any, context: any) => boolean;
    resolve?: (payload: any, args: any, context: any, info: any) => any;
}
```

## Minimal reproduction of the problem with instructions

See above.

## What is the motivation / use case for changing the behavior?

Without this change, you are essentially forced to type the resolve and filter function arguments as any.

## Environment
<pre><code>
Nest version: 6.2.4
</code></pre>

**Comment by kamilmysliwiec** [comment]
Thanks for reporting! Could you please check `@next` release? `$ npm i @nestjs/graphql@next`

**Comment by PachowStudios** [comment]
Filter looks good now, but resolve still has the same problem.

**Comment by PachowStudios** [comment]
I'm also going to hijack this issue to ask if you could please add a changelog specifically for @nestjs/graphql and @nestjs/typeorm. Currently it's difficult to know what changed in these auxiliary packages :)

**Comment by kamilmysliwiec** [comment]
Published as 6.2.3

> I'm also going to hijack this issue to ask if you could please add a changelog specifically for @nestjs/graphql and @nestjs/typeorm. Currently it's difficult to know what changed in these auxiliary packages :)

Definitely, something that we have to figure out as soon as possible :)
